api/marketdata.go

```diff
@@ -173,26 +173,39 @@ func getChartsHandler() func(c *gin.Context) {
 // @Param currency query string false "The currency to show coin info in" default(USD)
 // @Success 200 {object} watchmarket.ChartCoinInfo
 // @Router /v1/market/info [get]
-func getCoinInfoHandler() func(c *gin.Context) {
-	var charts = market.InitCharts()
+func getCoinInfoHandler(charts *market.Charts, ac assets.AssetClient) func(c *gin.Context) {
 	return func(c *gin.Context) {
 		coinQuery := c.Query("coin")
+		if len(coinQuery) == 0 {
+			ginutils.RenderError(c, http.StatusBadRequest, "No coin provided")
+			return
+		}
+
 		coinId, err := strconv.Atoi(coinQuery)
 		if err != nil {
-			ginutils.RenderError(c, http.StatusBadRequest, "Invalid coin")
+			ginutils.RenderError(c, http.StatusBadRequest, "Invalid coin provided")
 			return
 		}
 
 		token := c.Query("token")
 		currency := c.DefaultQuery("currency", watchmarket.DefaultCurrency)
 		chart, err := charts.GetCoinInfo(uint(coinId), token, currency)
-		if err != nil {
-			logger.Error(err, "Failed to retrieve coin info", logger.Params{"coin": coinId, "currency": currency})
+		if err == watchmarket.ErrNotFound {
+			ginutils.RenderError(c, http.StatusNotFound, fmt.Sprintf("Coin info for coin id %d (token: %s, currency: %s) not found", coinId, token, currency))
+			return
+		} else if err != nil {
```

Comment:
if err != nil 
there is no sense to have it here

Reply:
I don't understand?
Are you saying the `else` is redundant?